src/community/mapml/src/main/java/org/geoserver/mapml/MapMLController.java

```diff
@@ -137,7 +138,13 @@ public String Html(
         final String workspace;
         String styleName = style.orElse("");
         String imageFormat = format.orElse("image/png");
-        String baseUrl = ResponseUtils.baseURL(request);
+        String proxyBaseUrl = geoServer.getSettings().getProxyBaseUrl();
+        String baseUrl =
+                ResponseUtils.buildURL(
+                        proxyBaseUrl != null ? proxyBaseUrl : ResponseUtils.baseURL(request),
```

Comment:
Should still just use the ResponseUtils.baseURL(request), the proxy base URL should not be extracted from the config and used in this role, will work in some simple cases, but don't think it will do in general (e..g, when the parameter extractor plugin is in use). That said, just a suggestion, this is a community plugin, the only requirement is not to break the compile phase of the build (tests are not run, that's a privilege for supported modules).